mempool/mempool.go

```diff
@@ -91,6 +91,15 @@ type Mempool interface {
 	// trigger once every height when transactions are available.
 	EnableTxsAvailable()
 
+	// GetTxByKey gets a tx by its key from the mempool cache. Returns the tx and a bool indicating its presence in the tx cache.
+	// Used in the RPC endpoint: TxStatus.
+	GetTxByKey(key types.TxKey) (types.Tx, bool)
+
+	// IsTxEvicted returns true if the tx is evicted from the mempool and exists in the
```

Comment:
nit

```suggestion
	// IsTxEvicted returns true if the tx was evicted from the mempool and exists in the
```

Reply:
[question] what happens if the tx gets evicted from the mempool and then the user re-submits the exact same tx so that it exists in the tx cache? Is it possible for one tx to exist in both the tx cache and the evicted tx cache at the same time?

Reply:
yes, it's possible that a previously evicted tx exists in both caches. evicted tx only gets rid of its records once it reaches its max size(v0) or once it's reset alongside all other caches(v1). I think how we can avoid confusion between re-added txs is if we check for eviction last.

Reply:
I think a more correct phrasing of the method is `WasRecentlyEvicted`